samples/app-anonymous-users/csharp/ClientApp/src/components/shareview.jsx

```diff
@@ -84,11 +112,195 @@ const ShareView = () => {
+                return getServerSideTokenFb(result.idToken); // This method get a face book user profile details.
+            })
+            .catch((error) => {
+                console.log(error);
+            });
+    }
+
+    // Get client side token for facebook.
+    const fbAuthentication = () => {
+        var facebookAppId = process.env.REACT_APP_FACEBOOK_APP_ID;
+        let redirectUri = window.location.origin + "/facebook-auth-end";
+
+        return new Promise((resolve, reject) => {
+            microsoftTeams.authentication.authenticate({
+                url: `https://www.facebook.com/v12.0/dialog/oauth?client_id=${facebookAppId}&redirect_uri=${redirectUri}&state=1234535`,
+                width: 600,
+                height: 535
+            })
+                .then((result) => {
+                    let data = localStorage.getItem(result);
+                    let tokenDetails = JSON.parse(data);
+                    localStorage.removeItem(result);
+                    resolve(tokenDetails);
+                })
+                .catch((reason) => {
+                    reject(reason);
+                })
+        });
+    }
+
+    // Get face book user profile details.
+    const getServerSideTokenFb = (clientSideToken) => {
+        return new Promise(() => {
+            microsoftTeams.app.getContext().then(() => {
+                $.ajax({
+                    type: 'POST',
+                    url: '/getFbAccessToken',
+                    dataType: 'json',
+                    data: {
+                        'accessToken': clientSideToken,
+                    },
+                    success: function (responseJson) {
+                        let facebookProfile = JSON.parse(responseJson);
+                        // This variables will load the values to the labels
+                        setUserName("Welcome: " + facebookProfile.name);
+                        setdisabledfacebookBtn(false);
+                        setEnableVoteDiv(true);
+                    },
+                    error: function (textStatus, errorThrown) {
+                        console.log("textStatus: " + textStatus + ", errorThrown:" + errorThrown);
+                    }
+                });
+            });
+        });
+    }
+
+    // Tab sso authentication.
+    const ssoAuthentication = () => {
+        getClientSideToken()
+            .then((clientSideToken) => {
+                return getServerSideToken(clientSideToken);
+            })
+            .catch((error) => {
+                if (error === "invalid_grant") {
+                    // Display in-line button so user can consent
+                    IsSetVisibleConsent(true);
+                    ssoSetAuthenticationBtn(false);
+                } else {
+                    // Display in-line button so user can consent
+                    IsSetVisibleConsent(true);
+                    ssoSetAuthenticationBtn(false);
+                }
+            });
```

Comment:
Why do we need an if else, if the outcomes is the exact same?

Reply:
if (error === "invalid_grant") {
If we get any other error, we need to disable this button.

Reply:
But the outcome is the exact same irrespective of the error. 

If we get an invalid_grant,
we set `IsSetVisibleConsent` to `true`
and we set `ssoSetAuthenticationBtn` to `false`
Else if we get any other error 
We also set `IsSetVisibleConsent` to `true`
and we set `ssoSetAuthenticationBtn` to `false`

Reply:
Fixed